core/src/main/java/com/linecorp/armeria/client/cookie/CookieClient.java

```diff
@@ -79,12 +79,11 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
             req = req.withHeaders(req.headers().toBuilder().add(HttpHeaderNames.COOKIE, cookieHeader));
             ctx.updateRequest(req);
         }
-        return unwrap().execute(ctx, req).mapHeaders(headers -> {
+        return unwrap().execute(ctx, req).peekHeaders(headers -> {
```

Comment:
Q) Previously, `HttpResponse#mapHeaders` doesn't process informational headers.
Is there no need to preserve behavior?

https://github.com/line/armeria/blob/425953f35c2534255c798003120f80c5fec07f41/core/src/main/java/com/linecorp/armeria/common/HttpResponse.java#L757

Reply:
```java
default HttpResponse peekInformational(Consumer<? super ResponseHeaders> action) {
    requireNonNull(action, "action");
    final Consumer<? super ResponseHeaders> doWhenNonInformationalHeaders = headers -> {             
        if (!headers.status().isInformational()) {
            action.accept(headers);
        }
    };
    final StreamMessage<HttpObject> stream = peek(doWhenNonInformationalHeaders, ResponseHeaders.class);
    return of(stream);
}

default HttpResponse peekHeaders(Consumer<? super ResponseHeaders> action) {
    requireNonNull(action, "action");
    final Consumer<? super ResponseHeaders> doWhenInformationalHeaders = headers -> {             
        if (headers.status().isInformational()) {
            action.accept(headers);
        }
    };
    final StreamMessage<HttpObject> stream = peek(doWhenInformationalHeaders, ResponseHeaders.class);
    return of(stream);
}
```

If we need to split as `peekHeaders`, `peekInformational` here like HttpResponse#`mapHeaders`, `mapInformational`, 
I think I can do it like above! 😄

I'll follow your choice!

Reply:
How about not adding `peekInformational` and delaying adding until we really need it?
We don't provide a getter for informational headers in `SplitHttpResponse`. IIRC, it's because there are not so many chances that users want to use the informational headers. I think it also applies to `peekInformational`.

Reply:
> How about not adding peekInformational and delaying adding until we really need it?

+1

> Is there no need to preserve behavior?

We need to provide an identical behavior. Let's exclude informational headers from `peekHeaders`.

> `peekData` etc.. in the future..

As we provided `mapData`, I am leaning toward adding `peekData` and `peekTrailers`.

Reply:
> Let's exclude informational headers from peekHeaders.

I got it! I'll update PR 😄